src/v2/user-enrichment.service.ts

```diff
@@ -18,7 +29,51 @@ export class UserEnrichmentService {
   ) {}
 
   // TODO implementation goes here
-  enrichUsers(companyId: number, matchedUsers: MatchedUser[]) {
-    throw new Error("Method not implemented.");
+  async enrichUsers(companyId: number, matchedUsers: MatchedUser[]) {
+  // Call integration service and retrieve the external users
+   const externalUsers = await this.integrationService.getExternalEmployees(companyId);
+   const identifiableExternalUsers: IdentifiableExternalUser[] = externalUsers.map(externalUser => {
+     return {
+       ...externalUser,
+       identifier: `${externalUser.FamilyName} ${externalUser.GivenName}`.toLowerCase()
+     }
+   })
+
+  // Match the local users with the external users
+  //  TODO: Fix name for this variable
+    const externalUsers2: IdentifiableExternalUser[] = identifiableExternalUsers.map(externalUser => {
+      const matchedUser = matchedUsers.find(user => user.identifier.toLowerCase() === externalUser.identifier);
+      if (!matchedUser) return externalUser;
+      return {
+        ...externalUser, userId: matchedUser.userId
+      }
+    })
+
+  //  For each external user, upsert the user information with address
+    let addressesToUpdate: ExternalUserAddresses[] = [];
+
+   externalUsers2.forEach((externalUser) => {
+     // If could not match user
+     if (!externalUser.userId) return;
+
+     //  Convert external user address format to database format
+     const {PrimaryAddr: externalAddress} = externalUser
+
+      const userAddress: UserAddress = {
+        streetName: externalAddress.StreetLine,
+        city: externalAddress.City,
+        country: externalAddress.Country
+      }
+
+      addressesToUpdate.push({
+        userId: externalUser.userId,
+        address: userAddress
+      })
+    })
+
+  //  Update all information of address
+    for (const addressToUpdate of addressesToUpdate) {
+      await this.userAddressService.updateAddress(addressToUpdate.userId, addressToUpdate.address)
```

Comment:
This only updated the addresses, we should handle address creation too for the users that did not have an initial address.